### PR TITLE
*: add a restriction for smooth-upgrade-tidb

### PR DESCRIPTION
**Pull request by Oreoxmt**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v8.0 (TiDB 8.0 versions)
- [x] v7.6 (TiDB 7.6 versions)
- [x] v7.5 (TiDB 7.5 versions)
- [x] v7.4 (TiDB 7.4 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/16673
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by zimulala — APPROVED** [review]

File: smooth-upgrade-tidb.md [hunk 1]
```diff
@@ -84,6 +84,10 @@ When using the smooth upgrade feature, note the following limitations.
     * Run DDL operations on system tables (`mysql.*`, `information_schema.*`, `performance_schema.*`, and `metrics_schema.*`).
     * Manually cancel DDL jobs: `ADMIN CANCEL DDL JOBS job_id [, job_id] ...;`.
     * Import data.
+    * If the TiDB Distributed eXecution Framework (DXF) is enabled before the smooth upgrade, and there are ongoing `ADD INDEX` or `IMPORT INTO` tasks during the upgrade, perform the following steps to mitigate potential issues:
+      1. Disable the DXF by setting [`tidb_enable_dist_task`](/system-variables.md#tidb_enable_dist_task-span-classversion-marknew-in-v710span) to `OFF` (its default value).
```
Comment (Oreoxmt) [inline smooth-upgrade-tidb.md:88]:
```suggestion
      1. Disable the DXF by setting [`tidb_enable_dist_task`](/system-variables.md#tidb_enable_dist_task-new-in-v710) to `OFF` (its default value).
```

File: smooth-upgrade-tidb.md [hunk 2]
```diff
@@ -84,6 +84,10 @@ When using the smooth upgrade feature, note the following limitations.
     * Run DDL operations on system tables (`mysql.*`, `information_schema.*`, `performance_schema.*`, and `metrics_schema.*`).
     * Manually cancel DDL jobs: `ADMIN CANCEL DDL JOBS job_id [, job_id] ...;`.
     * Import data.
+    * If the TiDB Distributed eXecution Framework (DXF) is enabled before the smooth upgrade, and there are ongoing `ADD INDEX` or `IMPORT INTO` tasks during the upgrade, perform the following steps to mitigate potential issues:
```
Comment (Oreoxmt) [inline smooth-upgrade-tidb.md:87]:
```suggestion
    * Run distributed `ADD INDEX` or `IMPORT INTO` tasks. If the TiDB Distributed eXecution Framework (DXF) is enabled before the smooth upgrade, and there are ongoing `ADD INDEX` or `IMPORT INTO` tasks during the upgrade, perform the following steps to mitigate potential issues:
```

**Review by qiancai — APPROVED** [review]
Rest LGTM

File: smooth-upgrade-tidb.md [hunk 3]
```diff
@@ -75,7 +75,10 @@ When using the smooth upgrade feature, note the following limitations.
 
 ### Limitations on user operations
 
-* Before the upgrade, if there is a canceling DDL job in the cluster, that is, an ongoing DDL job is being canceled by a user, because the job in the canceling state cannot be paused, TiDB will retry canceling the job. If the retry fails, an error is reported and the upgrade is exited.
+* Before the upgrade, consider the following restrictions:
+
+    * If there is a canceling DDL job in the cluster, that is, an ongoing DDL job is being canceled by a user, because the job in the canceling state cannot be paused, TiDB will retry canceling the job. If the retry fails, an error is reported and the upgrade is exited.
+    * If the TiDB Distributed eXecution Framework (DXF) is enabled, disable it by setting [`tidb_enable_dist_task`](/system-variables.md#tidb_enable_dist_task-new-in-v710) to `OFF` (its default value) and ensure that all ongoing distributed `ADD INDEX` and `IMPORT INTO` tasks are completed. Alternatively, you can cancel these tasks and wait until the upgrade is complete to restart them. Otherwise, the `ADD INDEX` operations during the upgrade might cause data index inconsistency.
```
[the remark this answers is not in the document]
Comment (qiancai) [reply smooth-upgrade-tidb.md:81]: reason: 避免 Alternatively 看起来像是前面一整句话的 另一种办法

**Pull request merged**